matlab_simulation/08-planning/shortestpath.m

```diff
@@ -20,71 +49,224 @@
+                done = 1;
+                % Move best to closed set
+                C = [C; bestnode];                
+                continue;
+            end 
+            
+         case 4 %Dijkstra's 
+            % Check if open set is empty
+            if (length(OpenSet(:,1))==0)
+                spath = [];
+                sdist = 0;
+                return;
+            end             
+            [val, best] = min(OpenSet(:,3));
+            bestnode = OpenSet(best,:);
+            % Check end condition
+            if (bestnode(1)==finish)
+                done = 1;
+                % Move best to closed set
+                C = [C; bestnode];                
+                continue;
+            end           
 
-    % Check end condition
-    if (bestnode(1)==finish)
-        done = 1;
-        continue;
-    end
+            
+         case 2 %Breadth-First
+            if (length(OpenSet(:,1))==0)
+                done = 1;
+                continue;
+            end
+             best = 1;  % Grap next node in the open set
+             bestnode = OpenSet(best,:);
+            % remove best node from open set
+            OpenSet = OpenSet([best+1:end],:);                 
+            % Check end condition
+            if (bestnode(1)==finish)
+               % Move best to closed set
+                C = [C; bestnode];
+                continue;
+            end            
+         case 3 %Depth-First
+            if (length(OpenSet(:,1))==0)
+                done = 1;
+                continue;
+            end
+            best = 1;  % Grap next node in the open set
+            bestnode = OpenSet(best,:);
+            % remove best node from open set
+            OpenSet = OpenSet([best+1:end],:);            
+            % Check end condition
+            if (bestnode(1)==finish)
+               % Move best to closed set
+                C = [C;bestnode];
+                continue;
+            end  
+
+     end
+   
+     % Move best to closed set
+     C = [C; bestnode];
+
+    
 
     % Get all neighbours of best node
     neigh = find(edges(bestnode(1),:)==1);
     
     % Process each neighbour
     for i=1:length(neigh)
+        % If neighbour is in closed set, skip        
         found = find(C(:,1)==neigh(i),1);
         if (length(found)==1)
-            continue;
-        end
-        dtogo = norm(nodes(neigh(i),:)-nodes(finish,:));
-        dcur = bestnode(4)+dists(bestnode(1),neigh(i));
-        found = find(O(:,1)==neigh(i),1);
-        if (length(found)==0)
-            O = [O; neigh(i) bestnode(1) dtogo+dcur dcur]; 
-        else
-            if (dcur < O(found,4))
-                O(found,:) = [neigh(i) bestnode(1) dtogo+dcur dcur];
+            if(C(found,1)~=finish)
+                continue;
             end
         end
+        
+        dcur = bestnode(4)+dists(bestnode(1),neigh(i));
+        found = find(OpenSet(:,1)==neigh(i),1);
+        switch method
```

Comment:
I find three sets of switch used, maybe an explanation for each switch by at the previous line of the switch line a good way to tell the meaning.